pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -440,6 +445,138 @@ func validateRunStrategy(field *k8sfield.Path, spec *v1.VirtualMachineSpec) (cau
 	return causes
 }
 
+func validateLiveUpdateFeatures(field *k8sfield.Path, spec *v1.VirtualMachineSpec, config *virtconfig.ClusterConfig) (causes []metav1.StatusCause) {
```

Comment:
readability improvement:
IMHO this function looks too long.
How about splitting its content in subfuctions?
I'd expect `validateUpdateFeatures` to be a sort of wrapper:
```
func validateLiveUpdateFeatures(field *k8sfield.Path, spec *v1.VirtualMachineSpec, config *virtconfig.ClusterConfig) (causes []metav1.StatusCause) {
	if config.IsVmRolloutStrategyLiveUpdate() {
		if spec.Instancetype != nil {
			causes = append(causes, metav1.StatusCause{
				Type:    metav1.CauseTypeFieldValueNotSupported,
				Message: fmt.Sprintf("Cannot configure instance type when the vmRolloutStrategy is LiveUpdate"),
				Field:   field.Child("instancetype").String(),
			})
		}

		causes = append(causes, validateLiveUpdateCPU(field, &spec.Template.Spec.Domain)...)

		if hasCPURequestsOrLimits(&spec.Template.Spec.Domain.Resources) {
			causes = append(causes, metav1.StatusCause{
				Type:    metav1.CauseTypeFieldValueInvalid,
				Message: fmt.Sprintf("Configuration of CPU resource requirements is not allowed when CPU live update is enabled"),
				Field:   field.Child("template", "spec", "domain", "resources").String(),
			})
		}

		causes = append(causes, validateLiveUpdateMemory(field, &spec.Template.Spec.Domain, spec.Template.Spec.Architecture)...)
	}

	return causes
}
```
with `validateLiveUpdateCPU` and `validateLiveUpdateMemory`.

Commit example: https://github.com/fossedihelm/kubevirt/commit/2cdd7ef2af607ed8f1e92f081d7c23116723b700